crates/bevy_asset/src/loader.rs

```diff
@@ -165,6 +165,30 @@ impl<A: Asset> LoadedAsset<A> {
             meta,
         }
     }
+
+    /// Cast (and take ownership) of the [`Asset`] value of the given type. This will return [`Some`] if
+    /// the stored type matches `A` and [`None`] if it does not.
+    pub fn take(self) -> A {
+        self.value
+    }
+
+    /// Retrieves a reference to the internal [`Asset`] type, if it matches the type `A`. Otherwise returns [`None`].
```

Comment:
```suggestion
    /// Retrieves a reference to the internal [`Asset`] type.
```

I believe this docstring is out-of-date based on an optional version of the original API.